src/Navigation/Search/searchSaga.js

```diff
@@ -0,0 +1,19 @@
+import { put, call, debounce } from "redux-saga/effects";
+import { fetchDataSucces, fetchDataFailure, fetchData } from "./searchSlice";
+import { getSearch } from "./getSearch";
+
+function* fetchDataHandler(action) {
+    try {
+        yield call(fetchData);
+        const result = yield call(getSearch, action.payload.query);
+        yield put(fetchDataSucces({ data: result }));
+    } catch (error) {
+        yield put(fetchDataFailure({ error }));
+        console.log(error);
+    };
+};
+
+export function* watchFetchData() {
```

Comment:
I would use less generic name for your watcher, as all of them run on data fatching

Reply:
i changed this name  , now  this saga have : searchSaga :)